src/screens/FiatExchange/FiatCrypto.js

```diff
@@ -0,0 +1,216 @@
+import type { Accounts } from 'models/Account';
+import { Container, Wrapper } from 'components/Layout';
+import ErrorMessage from 'components/ErrorMessage';
+import Header from 'components/Header';
+import SelectorInput from 'components/SelectorInput';
+
+function SelectorInputTemplate(locals) {
+  const {
+    config: {
+      label,
+      hasInput,
+      wrapperStyle,
+      placeholderSelector,
+      placeholderInput,
+      options,
+      inputAddonText,
+    },
+  } = locals;
+  const errorMessage = locals.error;
+  const inputProps = {
+    onChange: locals.onChange,
+    onBlur: locals.onBlur,
+    keyboardType: locals.keyboardType,
+    maxLength: 42,
+    label,
+    placeholderSelector,
+    placeholder: placeholderInput,
+  };
+
+  return (
+    <SelectorInput
+      inputProps={inputProps}
+      options={options}
+      errorMessage={errorMessage}
+      hasInput={hasInput}
+      wrapperStyle={wrapperStyle}
+      value={locals.value}
+      inputAddonText={inputAddonText}
+    />
+  );
+}
+
+const { Form } = t.form;
+
+const FromOption = t.refinement(t.Object, ({ selector }) => {
+  return !!Object.keys(selector).length;
+});
+
+const SourceAsset = t.struct({
+  fromInput: FromOption,
+});
+
+const formOptions = {
+  fields: {
+    fromInput: {
+      keyboardType: 'decimal-pad',
+      template: SelectorInputTemplate,
+      config: {
+        label: 'Buying',
+        hasInput: true,
+        options: [],
+        placeholderSelector: 'select',
+        placeholderInput: '0',
+      },
+    },
+  },
+};
+
+type Props = {
+  wallet: Object,
+  navigation: NavigationScreenProp<*>,
+  assets: Object,
+  user: Object,
+  accounts: Accounts,
+};
+
+type State = {
+  error: string,
+  value: Object,
+  options: Object,
+};
+
+class FiatCrypto extends React.Component<Props, State> {
+  fiatCryptoForm = t.form;
+
+  state = {
+    error: '',
+    value: {
+      fromInput: {
+        selector: {},
+        input: '',
+      },
+    },
+    options: formOptions,
+  };
+
+  handleFormChange = (value: Object) => {
+    this.setState({ value });
+    this.updateOptions();
+  };
+
+  generateAssetsOptions = (assets) => {
+    const assetsList = Object.keys(assets).map((key: string) => assets[key]);
+    const alphabeticalAssets = assetsList.sort((a, b) => a.symbol.localeCompare(b.symbol));
+    return alphabeticalAssets.map(({ symbol, iconUrl, ...rest }) => {
+      return ({
+        key: symbol,
+        value: symbol,
+        icon: iconUrl,
+        iconUrl,
+        symbol,
+        ...rest,
+      });
+    });
+  };
+
+  updateOptions = () => {
+    const { assets } = this.props;
+
+    const optionsFrom = this.generateAssetsOptions(assets);
+
+    const newOptions = t.update(this.state.options, {
+      fields: {
+        fromInput: {
+          config: {
+            options: { $set: optionsFrom },
+          },
+        },
+      },
+    });
+
+    this.setState({ options: newOptions });
+  };
+
+  componentDidMount = () => {
```

Comment:
Minor: pls move this method above the `handleFormChange` as life-cycle methods normally go first

Reply:
d154ef1